CMakeLists.txt

```diff
@@ -47,6 +47,7 @@ option(CMAKE_POSITION_INDEPENDENT_CODE "Build static libraries with -fPIC" ON)
 option(BUILD_SHARED_LIBS "Build the library as a shared library" ON)
 
 option(SDL2IMAGE_INSTALL "Enable SDL2_image install target" ON)
+option(SDL2IMAGE_DEPS_SHARED "Load dependencies dynamically" ON)
```

Comment:
This is one of the things I was not sure about in the cmake build script:
does it make sense to dynamically load only certain dependencies?

Currently, this variable is used as a default variable for `cmake_dependent°option`.
Personally, I would ditch all `SDL2IMAGE_xxx_SHARED` options and only use this one variable.

Reply:
> does it make sense to dynamically load only certain dependencies?

I don't see why not: a Linux distribution that is less hardline about normal linking than Debian might want to make libjpeg and libpng be hard dependencies, but make libavif and libjxl weak/optional dependencies.